File: cs/sdk/src/Proton.Drive.Sdk/Nodes/DegradedFileNode.cs
using Proton.Sdk;

namespace Proton.Drive.Sdk.Nodes;

public sealed record DegradedFileNode : DegradedNode
{
    public required string MediaType { get; init; }

    public required DegradedRevision? ActiveRevision { get; init; }

    public required long TotalStorageQuotaUsage { get; init; }
}

File: kt/sdk/src/main/kotlin/me/proton/drive/sdk/entity/DegradedFileNode.kt
package me.proton.drive.sdk.entity

import java.time.Instant

data class DegradedFileNode(
    override val uid: NodeUid,
    override val parentUid: ParentNodeUid?,
    override val treeEventScopeId: ScopeId,
    override val name: Result<String>,
    val mediaType: String,
    override val creationTime: Instant,
    override val trashTime: Instant?,
    override val nameAuthor: Result<Author>,
    override val author: Result<Author>,
    val activeRevision: DegradedRevision?,
    val totalStorageQuotaUsage: Long,
    override val errors: List<DriveError>,
) : DegradedNode
